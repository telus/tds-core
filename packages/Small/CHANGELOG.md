# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 2.2.3 (2022-02-16)


### Bug Fixes

* gha release job npm publish token ([ce7aacd](https://github.com/telusdigital/tds/commit/ce7aacdd77e54c4f8e48eb31d0c05dc804530324))





## 2.2.2 (2022-02-15)


### Bug Fixes

* gha release job git config ([276ebb6](https://github.com/telusdigital/tds/commit/276ebb6968a0b56c9b87c178f6895a60ae108e71))





## 2.2.1 (2022-02-15)


### Bug Fixes

* gha release job git config ([1ab603d](https://github.com/telusdigital/tds/commit/1ab603d68c36219b0711fc353bc2515b64712ca9))





# [2.2.0](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.14...@tds/core-small@2.2.0) (2021-09-21)


### Features

* **core-small:** add react 17 to peer deps ([2b5338d](https://github.com/telusdigital/tds/commit/2b5338dd79999586cd773cc32b51fa0493704729))





## [2.1.14](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.13...@tds/core-small@2.1.14) (2021-01-15)

**Note:** Version bump only for package @tds/core-small





## [2.1.13](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.12...@tds/core-small@2.1.13) (2020-09-11)

**Note:** Version bump only for package @tds/core-small





## [2.1.12](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.11...@tds/core-small@2.1.12) (2020-09-08)


### Bug Fixes

* **core-small:** add support for styled-components v5 ([a718f4f](https://github.com/telusdigital/tds/commit/a718f4f9739ba431f3841be323d797cffc5bb548))





## [2.1.11](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.10...@tds/core-small@2.1.11) (2020-09-04)

**Note:** Version bump only for package @tds/core-small





## [2.1.10](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.9...@tds/core-small@2.1.10) (2020-02-03)

**Note:** Version bump only for package @tds/core-small





## [2.1.9](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.8...@tds/core-small@2.1.9) (2020-01-27)

**Note:** Version bump only for package @tds/core-small





## [2.1.8](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.7...@tds/core-small@2.1.8) (2020-01-23)

**Note:** Version bump only for package @tds/core-small





## [2.1.7](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.6...@tds/core-small@2.1.7) (2020-01-22)

**Note:** Version bump only for package @tds/core-small





## [2.1.6](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.5...@tds/core-small@2.1.6) (2019-10-09)

**Note:** Version bump only for package @tds/core-small





## [2.1.5](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.4...@tds/core-small@2.1.5) (2019-10-04)

**Note:** Version bump only for package @tds/core-small





## [2.1.4](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.3...@tds/core-small@2.1.4) (2019-09-06)

**Note:** Version bump only for package @tds/core-small





## [2.1.3](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.2...@tds/core-small@2.1.3) (2019-08-22)

**Note:** Version bump only for package @tds/core-small





## [2.1.2](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.1...@tds/core-small@2.1.2) (2019-08-09)

**Note:** Version bump only for package @tds/core-small





## [2.1.1](https://github.com/telusdigital/tds/compare/@tds/core-small@2.1.0...@tds/core-small@2.1.1) (2019-08-01)

**Note:** Version bump only for package @tds/core-small





# [2.1.0](https://github.com/telusdigital/tds/compare/@tds/core-small@2.0.2...@tds/core-small@2.1.0) (2019-07-02)


### Features

* **core-small:** add sup tag support ([461762b](https://github.com/telusdigital/tds/commit/461762b)), closes [#1059](https://github.com/telusdigital/tds/issues/1059)





## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-small@2.0.1...@tds/core-small@2.0.2) (2019-06-04)

**Note:** Version bump only for package @tds/core-small

## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-small@2.0.0...@tds/core-small@2.0.1) (2019-05-01)

**Note:** Version bump only for package @tds/core-small

# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.9...@tds/core-small@2.0.0) (2019-03-28)

### Features

- **core-small:** upgrade to styled-components ([3c45414](https://github.com/telusdigital/tds/commit/3c45414))

### BREAKING CHANGES

- **core-small:** upgrade to styled-components

## [1.0.9](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.8...@tds/core-small@1.0.9) (2019-03-20)

**Note:** Version bump only for package @tds/core-small

## [1.0.8](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.7...@tds/core-small@1.0.8) (2019-03-20)

**Note:** Version bump only for package @tds/core-small

## [1.0.7](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.6...@tds/core-small@1.0.7) (2019-03-18)

**Note:** Version bump only for package @tds/core-small

## [1.0.6](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.5...@tds/core-small@1.0.6) (2019-03-05)

**Note:** Version bump only for package @tds/core-small

## [1.0.5](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.4...@tds/core-small@1.0.5) (2019-02-06)

**Note:** Version bump only for package @tds/core-small

<a name="1.0.4"></a>

## [1.0.4](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.3...@tds/core-small@1.0.4) (2018-12-19)

**Note:** Version bump only for package @tds/core-small

<a name="1.0.3"></a>

## [1.0.3](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.2...@tds/core-small@1.0.3) (2018-12-06)

**Note:** Version bump only for package @tds/core-small

<a name="1.0.2"></a>

## [1.0.2](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.1...@tds/core-small@1.0.2) (2018-07-10)

**Note:** Version bump only for package @tds/core-small

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-small@1.0.0...@tds/core-small@1.0.1) (2018-06-14)

**Note:** Version bump only for package @tds/core-small
