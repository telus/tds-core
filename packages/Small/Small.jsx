import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../shared/utils/safeRest'

import styles from '../Text/Text.modules.scss'

/**
 * Small print, such as copyright and legal text.
 */
const Small = ({ children, ...rest }) => (
  <small {...safeRest(rest)} className={`${styles.small} ${styles.smallFont}`}>
    {children}
  </small>
)

Small.propTypes = {
  /**
   * The text.
   */
  children: PropTypes.string.isRequired,
}

export default Small
