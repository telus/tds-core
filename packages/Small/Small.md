```
<Small>
  Taxes and pay-per-use charges (including long distance, roaming and additional airtime or data) are extra.
</Small>
```
