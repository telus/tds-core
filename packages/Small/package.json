{
  "name": "@tds/core-small",
  "version": "2.1.3",
  "description": "Small print, such as copyright and legal text",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/shared-typography": "^1.3.3",
    "@tds/util-helpers": "^1.1.0",
    "prop-types": "^15.5.10"
  }
}
