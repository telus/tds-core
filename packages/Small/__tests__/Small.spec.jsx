import React from 'react'
import { shallow } from 'enzyme'

import Small from '../Small'

describe('Small', () => {
  const doShallow = props => shallow(<Small {...props}>Some content</Small>)

  it('renders', () => {
    const small = doShallow()

    expect(small).toMatchSnapshot()
  })

  it('renders an HTML small tag', () => {
    const small = doShallow()

    expect(small).toHaveDisplayName('small')
  })

  it('passes additional attributes to the small element', () => {
    const small = doShallow({ id: 'the-id' })

    expect(small).toHaveProp('id', 'the-id')
  })

  it('does not allow custom CSS', () => {
    const small = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(small).not.toHaveProp('className', 'my-custom-class')
    expect(small).not.toHaveProp('style')
  })
})
