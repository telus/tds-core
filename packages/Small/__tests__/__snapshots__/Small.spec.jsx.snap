// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small renders 1`] = `
<small
  className="small smallFont"
>
  Some content
</small>
`;
